packages/platforms/react-native/lib/retry-queue/file-based.ts

```diff
@@ -101,18 +102,22 @@ export default class FileBasedRetryQueue implements RetryQueue {
   }
 
   async flush (): Promise<void> {
-    const files = await this.directory.files()
+    this.requestQueue = this.requestQueue.then(async () => {
```

Comment:
Realised this request queueing was missing so calling `flush` multiple times in quick succession could deliver the same payload multiple times

Now it always waits for the previous `flush` to finish before starting another one